Rubberduck.Main/TestEngineProvider.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Rubberduck.UnitTesting;
+
+namespace Rubberduck
+{
+    public class TestEngineProvider
+    {
+        private _Extension _extension;
```

Comment:
There has to be another way to do this... having the entry point class as a dependency feels very backwards here.

Reply:
It is backwards and I was actually hoping you'd have suggestions on how to do this better. The problem is that `SetAddInObject` is called pretty early in the initialization process, so there isn't much to grab onto. Could it be moved further down the line without breaking things?